chess-AI/train.py

```diff
@@ -155,22 +154,24 @@ def training_episode(self, nnet, games, epochs, batch_size):
 
                         # Reset gradients
                         opt.zero_grad()
-
                 print(losses)
+            save_model(nnet)
```

Comment:
Could optionally have some flag that specifies whether we want to save the new model.
```py
if saving_model:
    save_model(<new_parameters>)
```

Reply:
Maybe even optionally two flags, one `saving_model` and one `save_path`:
```py
if save_path or saving_model:
    save_model(model=nnet, save_path=save_path)
```
This way, if `save_path` is `None`, but we are saving the model, we save it with a name based on the most recently saved model (as described above), otherwise we save it with the specified path.